### PR TITLE
T17886

### PR DESCRIPTION
**Pull request by smspillaz**
https://phabricator.endlessm.com/T17886

### CODE REVIEW

File: search-provider/eks-discovery-feed-provider.c [hunk 1]
```diff
@@ -1023,4 +1111,8 @@ eks_discovery_feed_database_content_provider_init (EksDiscoveryFeedDatabaseConte
   self->video_skeleton = eks_discovery_feed_video_skeleton_new ();
   g_signal_connect (self->video_skeleton, "handle-get-videos",
                     G_CALLBACK (handle_get_videos), self);
+
+  self->artwork_skeleton = eks_discovery_feed_artwork_skeleton_new ();
+  g_signal_connect (self->content_skeleton, "handle-article-card-descriptions",
```
Comment (erikos) [inline search-provider/eks-discovery-feed-provider.c]: That should be self->artwork_skeleton

Comment (smspillaz) [inline search-provider/eks-discovery-feed-provider.c]: doh, I knew I had stuffed this up.

**Comment by smspillaz** [comment]
Okay, rebased and fixed :)

**Comment by erikos** [comment]
Thanks Sam. The second commit with the refactor does still segfault though, that is the next one to fix...

**Comment by erikos** [comment]
This one looks ok to me too.  I have just seen traces of "Jul 31 16:18:51 endless eks-search-prov[3490]: g_variant_builder_open: assertion '!GVSB(builder)->expected_type || g_variant_type_is_subtype_of (type, GVSB(builder)->expected_type)' failed" in the logs which we fixed with https://github.com/endlessm/eos-knowledge-services/pull/21/commits/ca69f7a665ee6427dec08760057005778e70e38d - would be great to get this in.

**Pull request merged**